polars/polars-core/src/chunked_array/strings/encoding.rs

```diff
@@ -3,26 +3,63 @@ use {base64, hex};
 use crate::prelude::*;
 
 impl Utf8Chunked {
-    #[cfg(feature = "string_encoding")]
+    #[cfg(not(feature = "binary_encoding"))]
+    pub fn hex_decode(&self, strict: Option<bool>) -> PolarsResult<Utf8Chunked> {
+        let ca = self.apply_on_opt(|e| {
+            e.and_then(|s| {
+                hex::decode(s)
+                    // Safety
+                    // We already know that it is a valid utf8.
+                    .map(|bytes| Some(unsafe { String::from_utf8_unchecked(bytes) }.into()))
```

Comment:
Is every valid `utf8` also a valid `utf8` when we hex/base64 decode? I am not convinced of this.

I think we should do `from_utf8` so that utf8 validation runs.

Other than that we also should not heap allocate a new `String` but use a `str` instead.

And finally I think instead of `map().unwrap(None)` we can do `and_then` or `ok().and_then`.

Reply:
this is old code, I don't think it is valid either but I left it there as a fallback when binary isn't enabled in compilation

Reply:
fixed in last commit